server/workflowarchive/archived_workflow_server.go

```diff
@@ -291,12 +292,25 @@ func (w *archivedWorkflowServer) RetryArchivedWorkflow(ctx context.Context, req
 			return nil, sutils.ToStatusError(err, codes.Internal)
 		}
 
+		errCh := make(chan error, len(podsToDelete))
+		var wg sync.WaitGroup
+		wg.Add(len(podsToDelete))
 		for _, podName := range podsToDelete {
 			log.WithFields(log.Fields{"podDeleted": podName}).Info("Deleting pod")
-			err := kubeClient.CoreV1().Pods(wf.Namespace).Delete(ctx, podName, metav1.DeleteOptions{})
-			if err != nil && !apierr.IsNotFound(err) {
-				return nil, sutils.ToStatusError(err, codes.Internal)
-			}
+			go func(podName string) {
+				defer wg.Done()
+				err := kubeClient.CoreV1().Pods(wf.Namespace).Delete(ctx, podName, metav1.DeleteOptions{})
+				if err != nil && !apierr.IsNotFound(err) {
+					errCh <- err
+					return
+				}
+			}(podName)
```

Comment:
@agilgur5 do you have any opinions on this? My concern here is that this might very easily hit rate limitations. 
I see a similar PR has already been merged in though. 

I believe there are some timing related helper functions that can be used here, maybe that would help?  
I am concerned because I think some k8s platforms have relatively small QPS and for a large enough workflow, we might hit that easily.